processor/k8sattributesprocessor/internal/observability/observability.go

```diff
@@ -8,12 +8,150 @@ import (
 
 	"go.opencensus.io/stats"
 	"go.opencensus.io/stats/view"
+	"go.opentelemetry.io/collector/featuregate"
+)
+
+var renameInternalMetrics = featuregate.GlobalRegistry().MustRegister(
+	"processor.k8sattributes.RenameInternalMetrics",
+	featuregate.StageAlpha,
+	featuregate.WithRegisterDescription("When enabled, the k8sattributes processor will"+
+		" be renamed to new metric names"),
+)
+
+var (
+	mPodsUpdated        *stats.Int64Measure
+	mPodsAdded          *stats.Int64Measure
+	mPodsDeleted        *stats.Int64Measure
+	mPodTableSize       *stats.Int64Measure
+	mIPLookupMiss       *stats.Int64Measure
+	mNamespacesUpdated  *stats.Int64Measure
+	mNamespacesAdded    *stats.Int64Measure
+	mNamespacesDeleted  *stats.Int64Measure
+	mReplicaSetsUpdated *stats.Int64Measure
+	mReplicaSetsAdded   *stats.Int64Measure
+	mReplicaSetsDeleted *stats.Int64Measure
 )
 
 // TODO: re-think if processor should register it's own telemetry views or if some other
 // mechanism should be used by the collector to discover views from all components
 
-func init() {
+func Init() {
```

Comment:
Why not keep `init` here? Don't need to call it explicitly then

Reply:
Because if we use `init` function, the `View` object such as `viewPodsUpdated` should only be initialized as global variable, which results that the `stats.Measure` objects like `mPodsUpdated` are also  initialized as global variable, and the metric names are directly determined **before** parsing the new feature gate in the `main` package. And the name can not be changed after that.

So I think we should not use `init` function to initialize the metric objects such as `mPodsUpdated`, but use an exported function `Init`, which can be called after parsing the new feature gate in `main` package.  

@dmitryax WDYT?

Reply:
> which can be called after parsing the new feature gate in main package.

Why do we need that? The feature gate is initialized right here, not in main package.